Update how slider value is adjusted based on ticks positions

luislee818 opened the pull request:
When _ticks_positions_ are specified, the _raw_value_ calculated based on the percentage should not be adjusted with the _min_ value.

Changes:
* Do not adjust with base (min) after calculated with ticks positions
* Add example 13.a in index.html for the case above

seiyria commented:
Can you please provide a jsfiddle for the before and after (preferably with an explanation so this case is understood) as well as a unit test to cover this situation? Thanks!

rovolution commented on index.html:
```diff
@@ -711,6 +711,43 @@
       </code></pre>
   </div>
 
+  <div class='slider-example'>
```
these changes need to be added to the `index.tpl` file (which compile down to this file)

rovolution commented on js/bootstrap-slider.js:
```diff
@@ -206,7 +206,9 @@
 				toValue: function(percentage) {
-					var rawValue = percentage/100 * (this.options.max - this.options.min);
+					var rawValue = percentage/100 * (this.options.max - this.options.min),
+						shouldAdjustWithBase = true;
```
please declare this var on its own line

rovolution commented:
> Can you please provide a jsfiddle for the before and after (preferably with an explanation so this case is understood) as well as a unit test to cover this situation? Thanks!

Echoing @seiyria's sentiments...I'd like to see automated test coverage

luislee818 commented:
Thanks for the feedback.

About the tests, can I expose some objects in the closure for testing purpose? In this case, I would like to test the _toValue_ function on the _liner_ property of _SliderScale_ variable, as that's where the change I would like to make. Could _SliderScale_ be attached to _Slider_ variable, like this?
```javascript
Slider.Scale = SliderScale;
```

Or every test has to be done from the outside, on the DOM?

rovolution commented:
>> Or every test has to be done from the outside, on the DOM?

I would prefer DOM level assertions, simply because that is a better/more reliable test of the behavior

luislee818 commented:
Below are some examples showing the issue and the fix:

Version | Scenario (Tick starting from) | Result (dragging & clicking) | Demo
------------ | -------------| ------------- | -------------
v5.3.1 | 0 | working | http://codepen.io/luislee818/pen/NxqOWr
v5.3.1 | positive value | not working | http://codepen.io/luislee818/pen/XXmMpj
v5.3.1 | negative value | not working | http://codepen.io/luislee818/pen/vLNxgj
pull request | 0 | working | http://codepen.io/luislee818/pen/OMypWr
pull request | positive value | working | http://codepen.io/luislee818/pen/gPammQ
pull request | negative value | working | http://codepen.io/luislee818/pen/NxGppZ

I've added tests covering those three cases and updated implementation.

rovolution commented on test/specs/TickClickingBehaviorSpec.js:
```diff
@@ -0,0 +1,101 @@
+describe("TickClickingBehavior", function() {
+	var SLIDER_ID = "testSlider1";
```
this spec file is somewhat difficult to read.

Could you break each of these scenarios into their own explicit `it()` statement? I know it might be more repetition but it makes the file much more readable/easier to understand

luislee818 commented:
@rovolution I updated the spec.

rovolution commented:
@luislee818 thanks! Merging/bumping version

rovolution commented:
done! thanks again for the contribution @luislee818!
